kernel/posix/pthread_cond.c

```diff
@@ -33,7 +33,7 @@ static int cond_wait(pthread_cond_t *cv, pthread_mutex_t *mut, int timeout)
 	 */
 	pthread_mutex_lock(mut);
 
-	return ret == -EAGAIN ? -ETIMEDOUT : ret;
+	return ret == EAGAIN ? ETIMEDOUT : ret;
```

Comment:
The return code from _Swap() is a negative error code. Please revisit the error code conversion.